### PR TITLE
Ensure PersistedElement are unmounted on application logout

### PR DESCRIPTION
**Pull request by gsouquet**
Fixes vector-im/element-web#16725

`PersistedElement` is a React component that lets us float any widget on top of the app. It comes at the cost that we are responsible for its rendering. Upon a logout the React tree is correctly unmounted but no signal was sent to this class to notify it to remove its persisted elements

### CODE REVIEW

File: src/components/structures/MatrixChat.tsx [hunk 1]
```diff
@@ -586,6 +586,7 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
                 break;
             case 'logout':
                 dis.dispatch({action: "hangup_all"});
+                dis.dispatch({action: "logout"});
```
Comment (t3chguy) [inline src/components/structures/MatrixChat.tsx:589]: am I missing something or is a logout dispatch firing another logout dispatch which would get stuck in a loop?
Why is this needed atop the dispatcher bus listener for logout in PersistedElement itself?

Comment (gsouquet) [reply src/components/structures/MatrixChat.tsx]: You're completely right, that was the remainder of some of my testing that I forgot to clean up.
Thank you 💯

**Review by t3chguy — APPROVED** [review]
Great, thanks!
